src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -164,177 +146,77 @@ private static CopyColumnsTransform Create(IHostEnvironment env, ModelLoadContex
-            _schema = schema;
-            _columns = columns;
-            _colNewToOldMapping = new Dictionary<int, int>();
-            for (int i = 0; i < _columns.Length; i++)
+            protected override Delegate MakeGetter(IRow input, int iinfo, out Action disposer)
             {
-                if (!_schema.TryGetColumnIndex(_columns[i].Source, out int colIndex))
-                {
-                    throw _host.ExceptParam(nameof(schema), $"{_columns[i].Source} not found in {nameof(schema)}");
-                }
-                _colNewToOldMapping.Add(i, colIndex);
-            }
-        }
+                Host.AssertValue(input);
+                Host.Assert(0 <= iinfo && iinfo < _columns.Length);
+                disposer = null;
 
-        public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Action disposer)
-        {
-            _host.Assert(input.Schema == _schema);
-            var result = new Delegate[_columns.Length];
-            for (int i = 0; i < _columns.Length; i++)
-            {
-                if (!activeOutput(i))
-                    continue;
-                input.Schema.TryGetColumnIndex(_columns[i].Source, out int colIndex);
+                input.Schema.TryGetColumnIndex(_columns[iinfo].Source, out int colIndex);
                 var type = input.Schema.GetColumnType(colIndex);
-                result[i] = Utils.MarshalInvoke(MakeGetter<int>, type.RawType, input, colIndex);
+                return Utils.MarshalInvoke(MakeGetter<int>, type.RawType,  input, colIndex);
             }
-            disposer = null;
-            return result;
-        }
-
-        private Delegate MakeGetter<T>(IRow row, int src) => row.GetGetter<T>(src);
-
-        public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
-        {
-            var active = new bool[_schema.ColumnCount];
-            foreach (var pair in _colNewToOldMapping)
-                if (activeOutput(pair.Key))
-                    active[pair.Value] = true;
-            return col => active[col];
-        }
-
-        public RowMapperColumnInfo[] GetOutputColumns()
-        {
-            var result = new RowMapperColumnInfo[_columns.Length];
-            for (int i = 0; i < _columns.Length; i++)
-            {
-                _schema.TryGetColumnIndex(_columns[i].Source, out int colIndex);
-                var colType = _schema.GetColumnType(colIndex);
-                var meta = new RowColumnUtils.MetadataRow(_schema, colIndex, x => true);
-                result[i] = new RowMapperColumnInfo(_columns[i].Name, colType, meta);
-            }
-            return result;
-        }
 
-        public void Save(ModelSaveContext ctx)
-        {
-            _host.CheckValue(ctx, nameof(ctx));
-            ctx.CheckAtModel();
-            ctx.SetVersionInfo(GetVersionInfo());
+            private Delegate MakeGetter<T>(IRow input, int iinfo)
+                => input.GetGetter<T>(iinfo);
```

Comment:
declare the delegate inside the MakeGetter above, if it is the only thing that calls it.